### PR TITLE
Develop v1.2

### PR DESCRIPTION
**Pull request by uvolchyk**
v.1.2. in-app Web View
* Table view items will display Title and have annotations button.
* Clicking on annotations button will display details (title, description and date)
* Create a view to display the RSS item. The view should use WebView control so that rss item is displayed inside the application without going to the Safari browser.
* It should have toolbar with 5 buttons:
  * Back
  * Forward
  * Refresh
  * Stop
  * Launch in Safari

### CODE REVIEW

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+- (UIBarButtonItem *)reloadWebPageButton {
+    if(!_reloadWebPageButton) {
+        _reloadWebPageButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"refresh"]
+                                                         action:@selector(reload)] retain];
+    }
+    return _reloadWebPageButton;
+}
+
+- (UIBarButtonItem *)closeWebPageButton {
```
Comment (OrientCue) [inline rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:122]: Should be stop button (cancel load current url).
**Crash on tap.**

**Review by OrientCue — COMMENTED** [review]
Please fix crash.

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        _webView = [WKWebView new];
+        _webView.navigationDelegate = self;
+        _webView.translatesAutoresizingMaskIntoConstraints = NO;
+    }
+    return _webView;
+}
+
+- (UIBarButtonItem *)goBackButton {
+    if(!_goBackButton) {
+        _goBackButton = [[self webBarButtonItemWithImage:[UIImage imageNamed:@"arrow-narrow-left"]
```
Comment (OrientCue) [inline rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m]: Please make category for UIImage. 
NSString literal should be moved to global constant.

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                           completionHandler:^(BOOL success) {
+        [self closeWebPage];
+    }];
+}
+
+// MARK: - WKNavigationDelegate
+
+- (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction
+decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
+    switch (navigationAction.navigationType) {
```
Comment (OrientCue) [inline rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:161]: What is purpose of this method?

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m (hunk 2 above)
Comment (trotnic) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:97]: nice point, done

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m (hunk 3 above)
Comment (uvolchyk) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:161]: without implementing this method of delegate webview doesn't know how to deal with hyperlinks 
[read the docs pls](https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455641-webview?language=objc)

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m (hunk 1 above)
Comment (uvolchyk) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:122]: fixed

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m (hunk 2 above)
Comment (uvolchyk) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m]: nice point, done

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m (hunk 3 above)
Comment (OrientCue) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:161]: This method isn't mandatory to implement. Web view opens links without implementing it. You did not introduce any decide policy inside - in any case you call `decisionHandler(WKNavigationActionPolicyAllow);`

Comment (uvolchyk) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:161]: idk, without it links don't work

Comment (OrientCue) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:161]: Well I think that actually links are works, but there is no indication of loading process of webpage, and it seems as not working(no progress, navigation arrows didn'n change state, some kind of activity indicator etc). I pulled changes from branch and now can't check this:)

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m (hunk 2 above)
Comment (OrientCue) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m]: Resolved.

File: rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m (hunk 3 above)
Comment (OrientCue) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:161]: Resolved. It fixes opening any links which have `target="_blank"` attribute in their HTML <`a href`> tag.

Comment (uvolchyk) [reply rss-reader/Presentation/ChannelFeed/View/WebView/UVFeedItemWebViewController.m:161]: wow 😅✋

**Pull request merged**